src/mscorlib/src/System/String.cs

```diff
@@ -666,6 +666,37 @@ private unsafe String CtorCharPtrStartLength(char* ptr, int startIndex, int leng
         [MethodImplAttribute(MethodImplOptions.InternalCall)]
         public extern String(char c, int count);
 
+        private unsafe string CtorCharRefLength(ref char value, int length)
+        {
+            Debug.Assert(length >= 0);
+
+            if (length == 0)
+            {
+                return Empty;
+            }
+
+            try
+            {
+                string result = FastAllocateString(length);
+                fixed (char* dest = &result._firstChar, src = &value)
+                {
+                    wstrcpy(dest, src, length);
+                }
+                return result;
+            }
+            catch (NullReferenceException)
+            {
+                throw new ArgumentOutOfRangeException(nameof(value), SR.ArgumentOutOfRange_PartialWCHAR);
+            }
+        }
+
+        public String(ReadOnlySpan<char> value) :
```

Comment:
> Hmm. It seemed to work, in that I was able to successfully run all of the tests

You just got lucky that all tests passed. The GC tracking is wrong among other things: I have written a tests that calls the new string constructor and stresses the GC: https://gist.github.com/jkotas/6c92c0e432fa54a16a1a96f5e4528122. It fails with on checked CoreCLR:

```
Assert failure(PID 24632 [0x00006038], Thread: 27628 [0x6bec]): supportStringConstructors

CORECLR! MethodDesc::ReturnsObject + 0x32B (0x00007ffd`e7869bf7)
CORECLR! GetReturnKindFromMethodTable + 0xEB (0x00007ffd`e754a7f3)
CORECLR! GetReturnKind + 0x8D (0x00007ffd`e754a6a5)
```

> You think I should change it?

I think so.